### PR TITLE
Feature/account exists page

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- Update `validateUserRegistrationRequest` to resolve a boolean that indicates if account already exists (true) or more info is required (false).
- Add new step for "Account Already Created" page which allows for custom success screen content or strings

### CODE REVIEW

**Comment by emclaug2** [comment]
https://pxblue-temp.web.app/auth/register/invite?code=DEADBEEF will make a user perform the unregistered account workflow.  

https://pxblue-temp.web.app/auth/register/invite?code=PXWHITE will skip account registration since the app thinks the account exists already.

File: login-workflow/example/src/app/pages/auth/auth.component.ts [hunk 1]
```diff
@@ -70,7 +73,11 @@ import { Router } from '@angular/router';
 
         <!-- Custom Create Account page -->
         <ng-template #createAccountViaInvitePage>
-            <pxb-create-account-invite #createAccountInviteVC [accountDetails]="accountDetails">
+            <pxb-create-account-invite
+                #createAccountInviteVC
+                [accountDetails]="accountDetails"
+                [existingAccountSuccessScreen]="registrationSuccessScreen"
```
Comment (joebochill) [inline login-workflow/example/src/app/pages/auth/auth.component.ts:79]: It's a little confusing having these both use the same success screen. It makes it hard to tell if I'm seeing the existing account success screen or the regular one.

**Review by joebochill — CHANGES_REQUESTED** [review]
The Self-Registration flow is still asking me for account details even when I am resolving true in the registration service:
image

File: login-workflow/src/pages/create-account/steps/existing-account/existing-account.component.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+::ng-deep .pxb-account-existing-empty-state .pxb-empty-state-description {
+    color: inherit !important;
```
[the remark this answers is not in the document]
Comment (emclaug2) [reply login-workflow/src/pages/create-account/steps/existing-account/existing-account.component.scss:2]: I'll update the angular component lib used in the demo app to see if this style can be removed.

**Comment by emclaug2** [comment]
> The Self-Registration flow is still asking me for account details even when I am resolving true in the registration service

Good catch, I forgot the `validateUserRegistrationRequest` was actually called in the self-registration workflow too, just later on.  I'll add this feature in.

File: login-workflow/example/src/app/pages/auth/auth.component.ts (hunk 1 above)
Comment (dev-ptera) [reply login-workflow/example/src/app/pages/auth/auth.component.ts]: There's now a unique success screen for whenever an account has been previously registered.  

Currently, the 'create-account' 'already-registered' success screen shows the default.  
The 'create-account-invite' 'already-registered' success screen shows the customized one.  

The updates are hosted here: https://pxblue-temp.web.app/

Comment (emclaug2) [reply login-workflow/example/src/app/pages/auth/auth.component.ts]: There's now a unique success screen for whenever an account has been previously registered.  

Currently, the 'create-account' 'already-registered' success screen shows the default.  
The 'create-account-invite' 'already-registered' success screen shows the customized one.  

The updates are hosted here: https://pxblue-temp.web.app/

**Comment by huayunh** [comment]
Might need some word tweak on that screen. 

“It looks like the account created with email xxx already exists. Please click continue to log in.

If you think this is a mistake, please contact xxxx.”

feel free to change the wording

File: login-workflow/src/pages/create-account/steps/existing-account/existing-account.component.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (emclaug2) [reply login-workflow/src/pages/create-account/steps/existing-account/existing-account.component.scss:2]: > “It looks like the account created with email xxx already exists. Please click continue to log in.
> 
> If you think this is a mistake, please contact xxxx.”

I like this wording, I'll update the example.

**Comment by emclaug2** [comment]
I kept the wording the same and redployed the app in firebase for viewing.

**Review by joebochill — CHANGES_REQUESTED** [review]
Doing self-registration with an 'existing account' and custom success screen defined still takes me to the default success screen.

File: login-workflow/example/src/app/pages/auth/auth.component.ts [hunk 3]
```diff
@@ -59,6 +62,42 @@ import { Router } from '@angular/router';
...
+                <div style="margin: 0 24px">
+                    <div *ngIf="firstName" class="mat-display-1" style="margin-bottom: 24px">
+                        Welcome, PX White User
+                    </div>
+                    <div *ngIf="!firstName" class="mat-display-1" style="margin-bottom: 24px">Welcome!</div>
+                    <div class="mat-h4">
+                        It looks like the account created with email xxx already exists. Please click continue to log
+                        in.
+                    </div>
+                    <div class="mat-h4">If you think this is a mistake, please contact xxx</div>
```
[the remark this answers is not in the document]
Comment (emclaug2) [reply login-workflow/example/src/app/pages/auth/auth.component.ts:92]: I can adjust the wording just a little bit so it doesn't feel like it possibly read as an error message.

**Comment by joebochill** [comment]
> Doing self-registration with an 'existing account' and custom success screen defined still takes me to the default success screen.

This was my bad. I didn't realize that the custom screen wasn't hooked up in the example...works as expected.

**Review by joebochill — APPROVED** [review]

**Pull request merged**